### DISCUSSION TITLE
Installation issues with RConv on local machine

### DISCUSSION THREAD

**Discussion by vikaskurapati** [SeisSol/SeisSol#830] [General]
Hi,

I am trying to install rconv on my local machine. I believe that Proj has been rightly installed from https://github.com/SeisSol/SeisSol/tree/master/preprocessing/science/rconv#installing-proj4 and when I try to install rconv like mentioned in the above link, cmake throws the following warning: 
`CMake Warning at /usr/share/cmake-3.22/Modules/FindHDF5.cmake:747 (message):
  HDF5 found for language C is not parallel but previously found language is
  parallel.
Call Stack (most recent call first):
  CMakeLists.txt:66 (find_package)
`
I think this is somehow linked to the error I am trying to deal with when I try to compile with make. The error is as follows:
`/usr/bin/ld: /home/tools/lib/libhdf5.a(H5Zdeflate.o): undefined reference to symbol 'inflateEnd'
/usr/bin/ld: /lib/x86_64-linux-gnu/libz.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/SeisSol-rconv.dir/build.make:172: rconv] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/SeisSol-rconv.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
`
I have used `--enable-parallel` flag when compiling hdf5 and I am not sure why rconv cmake mentions that hdf5 found is not parallel. Is there a way to deal with this issue without trying to reinstall hdf5? Or is the problem to hdf5 in the first place? 

Thank you very much in advance

**Comment by sebwolf-de** [comment]
Hi, `inflateEnd` seems to be part of zlib: https://refspecs.linuxbase.org/LSB_3.0.0/LSB-Core-generic/LSB-Core-generic/zlib-inflateend-1.html
Can you check `your/build/directory/CMakeFiles/SeisSol-rconv.dir/link.txt`, whether it contains any reference to zlib? Maybe you need to add a `-lz` somewhere.

**Reply by sebwolf-de** [reply]
FYI: On my local machine, I have installed hdf5 with: 
```
spack install hdf5@1.10.7%gcc@12.2.1~cxx~fortran+hl~ipo~java+mpi+shared~szip~threadsafe+tools api=default build_system=cmake build_type=RelWithDebInfo patches=2a1e311 
```
Maybe that helps...

**Reply by vikaskurapati** [reply]
Hi, `link.txt` does not seem to have a reference to zlib. Do I need to add -lz flag while building hdf5 or while building rconv?

**Reply by krenzland** [reply]
I guess while building rconv. However, I'm a bit surprised that this is necessary, as the HDF5 linker flags should already include that. rconv and SeisSol seem to use the same linker flags for HDF5.

**Reply by krenzland** [reply]
Specifying "-lz" in the linker flags solved it.
